pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -511,50 +521,31 @@ func testHTTPSClassification(t *testing.T, tr *Tracer, clientHost, targetHost, s
 				extras:        make(map[string]interface{}),
 			},
 			preTracerSetup: func(t *testing.T, ctx testContext) {
-				cmd := testutil.HTTPPythonServer(t, ctx.serverAddress, testutil.Options{
-					EnableKeepAlive: false,
-					EnableTLS:       true,
-				})
+				cmd := gotlstestutil.NewGoTLSServer(t, ctx.serverAddress)
 				ctx.extras["cmd"] = cmd
 			},
 			postTracerSetup: func(t *testing.T, ctx testContext) {
 				cmd := ctx.extras["cmd"].(*exec.Cmd)
-				utils.WaitForProgramsToBeTraced(t, "shared_libraries", cmd.Process.Pid)
-				client := nethttp.Client{
+				goTLSAttachPID(t, cmd.Process.Pid)
+				t.Cleanup(func() {
+					goTLSDetachPID(t, cmd.Process.Pid)
+				})
+
+				client := &nethttp.Client{
 					Transport: &nethttp.Transport{
 						TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
 						DialContext:     defaultDialer.DialContext,
 					},
 				}
 
-				// Ensure that we see HTTPS requests being traced *before* the actual test assertions
-				// This is done to reduce test test flakiness due to uprobe attachment delays
-				require.Eventually(t, func() bool {
-					resp, err := client.Get(fmt.Sprintf("https://%s/200/warm-up", ctx.targetAddress))
-					if err != nil {
-						return false
-					}
-					_, _ = io.Copy(io.Discard, resp.Body)
-					_ = resp.Body.Close()
-
-					httpData := getConnections(t, tr).HTTP
-					for httpKey := range httpData {
-						if httpKey.Path.Content.Get() == resp.Request.URL.Path {
-							return true
-						}
-					}
+				// The server might not be ready to accept connection just yet, so we
+				// wait until it starts accepting them.
+				warmupURL := fmt.Sprintf("https://%s/200/warmup", ctx.targetAddress)
+				require.EventuallyWithT(t, func(c *assert.CollectT) {
+					makeRequest(c, client, warmupURL)
+				}, 5*time.Second, 100*time.Millisecond)
 
-					return false
-				}, 5*time.Second, 100*time.Millisecond, "couldn't detect HTTPS traffic being traced (test setup validation)")
-
-				t.Log("run 3 clients request as we can have a race between the closing tcp socket and the http response")
-				for i := 0; i < 3; i++ {
-					resp, err := client.Get(fmt.Sprintf("https://%s/200/request-1", ctx.targetAddress))
-					require.NoError(t, err)
-					_, _ = io.Copy(io.Discard, resp.Body)
-					_ = resp.Body.Close()
-					client.CloseIdleConnections()
-				}
+				makeRequest(t, client, fmt.Sprintf("https://%s/200/request-1", ctx.targetAddress))
```

Comment:
I dont see any reason to split the requests for warm-up and "post-warm-up"
in the test you are checking we classified the connection as HTTP + TLS, without adding any further verification (like client port, or the path). So, we might classify the first (or 10th) warm-up connection instead of the "actual" request